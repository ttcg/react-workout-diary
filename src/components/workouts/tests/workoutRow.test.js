import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WorkoutRow from '../workoutRow';

it('renders without crashing', () => {
    const props = {
        showEdit: jest.fn(),
        handleDelete: jest.fn(),
        item: {
            "id": '38046275-fe56-414e-9c51-75e2fc84438e',
            "workoutType": "Running",
            "date": "2019-01-02",
            "calories": 340
        }
    };

    const component = shallow(<WorkoutRow {...props} i={1}  />);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();

    component.find('Button.testEdit').simulate('click');
    expect(props.showEdit).toHaveBeenCalledTimes(1);

    component.find('Button.testDelete').simulate('click');
    expect(props.handleDelete).toHaveBeenCalledTimes(1);
});