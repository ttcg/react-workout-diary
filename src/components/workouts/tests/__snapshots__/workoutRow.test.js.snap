// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<tr>
  <th
    scope="row"
  >
    2
  </th>
  <td>
    <t
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="DD MMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    >
      2019-01-02
    </t>
  </td>
  <td>
    Running
  </td>
  <td>
    340
  </td>
  <td>
    <Button
      className="testEdit"
      color="link"
      onClick={[Function]}
      tag="button"
    >
      Edit
    </Button>
     |
    <Button
      className="testDelete"
      color="link"
      onClick={[Function]}
      tag="button"
    >
      Delete
    </Button>
  </td>
</tr>
`;
